pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,401 @@
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          KeyError: If the namespace is not defined in the repository.
+          ValueError: if the namespace is None.
+        """
+        pass
+
+    @abstractmethod
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the repository.
+
+        The namespace must not yet exist in the CIM repository.
+
+        The default connection namespace is automatically added to
+        the CIM repository upon creation of this connection.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          ValueError: If the namespace argument is None or the
+           namespace already exists.
+        """
+        pass
+
+    @abstractmethod
+    def remove_namespace(self, namespace):
+        """
+        Remove a CIM namespace from the repository.
+
+        The namespace must exist in the repository and must be empty.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          ValueError: Namespace argument is None or the repository namespace
+            is not empty
+          KeyError:  The namespace does not exist in the CIM repository.
+        """
+        pass
+
+    @abstractmethod
+    def get_class_repo(self, namespace):
+        """
+        Get the handle for the data CIM class object store for the namespace
+        provided.
+
+        Parameters:
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Returns:
+
+          Returns the instance of :class:`~pywbem_mock.InMemoryObjectStore` for
+          CIM classes which defines the methods exists(name), get(name)
+          create(), etc. for accessing the data in this store.
+
+        Raises:
+
+          ValueError: Namespace argument is None.
+          KeyError: If the namespace does not exist in the repository
+        """
+
+        pass
+
+    @abstractmethod
+    def get_instance_repo(self, namespace):
+        """
+        Get the handle for the data CIM instance object store for the namespace
+        provided.
```

Comment:
For the reasons stated further up, I suggest to change this to "Return the CIM instance object store ...".

Reply:
DONE